include/pet.h

```diff
@@ -267,6 +268,10 @@ struct pet_model{
 
   struct pet_bmi bmi;
 
+  // serialization data
+  char* serialized;
+  uint64_t serialized_length;
```

Comment:
This turns out to be a problem later.  I'll explain more in another part of the code, but a BMI function (`Get_var_nbytes`) needs to put this value into an integer, and it might be too big.  I haven't traced though how much this might impact the entire serialization implementation just yet.

Reply:
We actually use `GetValue("serialization_size")` to get it in full 64-bit range. I took `nbytes` returning `int` into account when designing this protocol.